Add bullet example.

binji opened the pull request:
This uses the bullet library build from naclports, as well as John
McCutchan's NaCl Acceleration Module repository (with some small
changes).

I've added a top-level Makefile that allows the same commands as the
other examples (e.g. "make TOOLCHAIN=pnacl") but calls into a shell
script to do the work.

binji committed:
Add bullet example.

This uses the bullet library build from naclports, as well as John
McCutchan's NaCl Acceleration Module repository (with some small
changes).

I've added a top-level Makefile that allows the same commands as the
other examples (e.g. "make TOOLCHAIN=pnacl") but calls into a shell
script to do the work.

binji commented:
I was thinking it might be better to copy some of the examples/bullet/data/\* files from the NaClAMBase repo as part of the build -- but it's also nice to have them checked in for inspection.

Thoughts?

ncbray commented:
I rather see unmodified files copied in - unless you suspect they may need to be modified in the future.  It makes it easier for me to review, if nothing else.

Windows also needs to be dealt with in some way (such as detecting and disabling?)

binji commented:
OK, moved all the changes to my own repo (https://github.com/binji/NaClAMBase), copy data files from that repo into the directory on build, and added an error when building on Windows. PTAL

binji commented:
- Added README.rst
- Rolled in fixes for building Bullet/naclports on Mac OS X
- Added Bullet to examples.html (behind the developer mode)

binji commented:
- Added loadnacl.js
- Added stylesheet
- tested with pepper_31 on emscripten + pnacl
